### PR TITLE
feat: add ts-check, lint, husky and pr template

### PR DESCRIPTION
**Pull request by alexCanditech**
## Description

**Asana:** https://app.asana.com/0/1203249289331793/1205677325919877/f

### Summary

This PR adds the following:

- ts-check
- lint checks
- husky pre-push and pre-commit scripts

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,20 +1,6 @@
...
-    '@typescript-eslint'
-  ],
-  include: [
-    './gtm.d.ts'
-  ],
   extends: [
-    'plugin:import/recommended',
-    'plugin:import/typescript',
-    'eslint:recommended',
+    '../web-app/.eslintrc.js',
```
Comment (alexCanditech) [inline .eslintrc.js:3]: @KnaanBrazCanditech @eyalCanditech @HaimCandiTech  I am not sure about this dependency but looked counterproductive to have separate configs, we should extract to a shared one IMO

File: .eslintrc.js [hunk 2]
```diff
@@ -1,20 +1,6 @@
...
-    '@typescript-eslint'
-  ],
-  include: [
-    './gtm.d.ts'
-  ],
   extends: [
-    'plugin:import/recommended',
-    'plugin:import/typescript',
-    'eslint:recommended',
+    '../web-app/.eslintrc.js', // TODO alexCanditech extract common rules to a shared config
```
Comment (alexCanditech) [inline .eslintrc.js:3]: @KnaanBrazCanditech @eyalCanditech @HaimCandiTech  I am not sure about this dependency but looked counterproductive to have separate configs, we should extract to a shared one IMO

Comment (eyalCanditech) [reply .eslintrc.js:3]: e2e should remain independent, so please copy and paste for now.

Comment (alexCanditech) [reply .eslintrc.js]: done, ready for re-review

**Review by KnaanBrazCanditech — APPROVED** [review]

**Pull request merged**